src/app-layer-htp-file.c

```diff
@@ -83,9 +83,9 @@ int HTPFileOpen(HtpState *s, HtpTxUserData *tx, const uint8_t *filename, uint16_
  * @param[in] rawvalue
  * @param[out] range
  *
- * @return HTP_OK on success, HTP_ERROR on failure.
+ * @return HTP_STATUS_OK on success, HTP_STATUS_ERROR on failure.
  */
-int HTPParseContentRange(bstr *rawvalue, HTTPContentRange *range)
+int HTPParseContentRange(const bstr *rawvalue, HTTPContentRange *range)
```

Comment:
adding `const` may be a separate commit

Reply:
Understood. There was a [previous discussion](https://github.com/OISF/suricata/pull/8335#discussion_r1063252451) on this as well. 

I would like to save the commit refactoring for after everything else is done. It's actually quite a lot of work to try to separate the commits and keep everything compiling at the same time, and doing it every time for v8, v9, v10, ... means redoing it every time.

Reply:
I am not sure what you mean by prerequisite commits? You mean just the const changes? Everything else actually requires the library to be present.

Reply:
I indeed mean everything that does not need the new library, (but that the new library may require)

Also removal of the unused soft limit value ?
You know better what else there may be :-)